src/main/java/folk/sisby/tinkerers_smithing/mixin/MixinRegistrySyncManager.java

```diff
@@ -0,0 +1,29 @@
+package folk.sisby.tinkerers_smithing.mixin;
+
+import folk.sisby.tinkerers_smithing.TinkerersSmithing;
+import net.fabricmc.fabric.impl.registry.sync.RegistrySyncManager;
+import net.minecraft.registry.Registry;
+import net.minecraft.util.Identifier;
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import java.util.Objects;
+
+@Mixin(RegistrySyncManager.class)
+public abstract class MixinRegistrySyncManager {
+	@Redirect(method = "createAndPopulateRegistryMap", at = @At(value = "INVOKE", target = "Lnet/minecraft/registry/Registry;getId(Ljava/lang/Object;)Lnet/minecraft/util/Identifier;"))
```

Comment:
can this be achieved without a redirect? possibly using a different target or a mixinextras annotation?

Reply:
Figured out what I think is a cleaner method